loaders/src/glTF/2.0/babylon.glTFLoader.ts

```diff
@@ -80,20 +80,38 @@ module BABYLON.GLTF2 {
         }
 
         // IE 11 Compatibility.
-        private static _createProgressEvent: (lengthComputable: boolean, loaded: number, total: number) => ProgressEvent =
-            (typeof (<any>window)["ProgressEvent"] === "function")
-                ? (lengthComputable, loaded, total) => {
+        private static _createProgressEvent = GLTFLoader._getCreateProgressEventFunction();
+        private static _getCreateProgressEventFunction(): (lengthComputable: boolean, loaded: number, total: number) => ProgressEvent {
+            // Use ProgressEvent if available.
+            if (typeof (<any>window)["ProgressEvent"] === "function") {
+                return (lengthComputable, loaded, total) => {
                     return new ProgressEvent("GLTFLoaderProgress", {
                         lengthComputable: lengthComputable,
                         loaded: loaded,
                         total: total
                     });
-                }
-                : (lengthComputable, loaded, total) => {
+                };
+            }
+
+            // Fallback to initProgressEvent if available.
+            const event = document.createEvent("ProgressEvent");

```

Comment:
we want to make sure it works event if document does not exist

Reply:
what do you recommend if document doesn't exist?

Reply:
we should not rely on DOM element :(